src/main/java/org/tikv/common/operation/iterator/RawScanIterator.java

```diff
@@ -67,11 +68,18 @@ TiRegion loadCurrentRegionToCache() throws GrpcException {
     }
   }
 
-  private boolean notEndOfScan() {
-    return limit > 0
-        && !(processingLastBatch
-            && (index >= currentCache.size()
-                || Key.toRawKey(currentCache.get(index).getKey()).compareTo(endKey) >= 0));
+  private boolean eof() {
```

Comment:
usually, `eof` is short for `end of file`, how about renaming it to `endOfScan()` to eliminate misunderstanding?

Reply:
Done.